Flexible featured image name

Agos95 opened the pull request:
### Purpose

This allows more flexibility in the naming of the featured image for posts, and fixes #2777.

With this change, featured image is searched following this order of operations:

1. Search for a file `*featured*` in the post directory
    - this is for backward compatibility
2. Search for a file `.Params.image.filename` in the post directory
    - this is for who prefers to have all the files related to a specific post inside the post directory
3. Search for a file `.Params.image.filename` in the `assets/media/` directory
     - this allows to use an image in `assets/media/` as a featured one. In this way, multiple posts can have the same featured figure without the need to upload the same image multiple times in multiple folders
     - furthermore, now posts can just be single markdown pages and they are not required to be page boundles (since the featured image is searched in `assets/media/` too)
     - in addition, using Hugo cascade, it is possible to specify a default featured image in config.yaml, which can be used as fallback in the case nothing is specified in the post front matter. The lines to add are:  
```yaml
cascade:
  # Configure blog posts (except post/_index.md)
  - _target:
      path: /post/[!_index]**
    image:
      # in assets/media/
      filename: default_featured_image.jpg
```

This PR also  #fixes the code in the different views in order to use the same logic to select the featured image.  
Lastly, the change is also applied in the other files where the featured image was searched by using the hard coded `*featured*` name.

#### Further CMS enhancement

With this PR, now the CMS can accept images with any name (thus fixing #2777). However, from the CMS you are not able to select an image which is already present in your `assets/media/`, neither you are able to upload an image to that folder, since the "image" widget is configured to look inside the post directory. 

Maybe it can be useful to open a new issue to address also this problem.

gcushen reviewed (changes requested):
Thanks for proposing an improvement!

1. Please move all the repetitive code into a single function so that there is one place to maintain this code going forward rather than pasting the same block of code into 8 files.
2. What's the purpose of adding `<!--unformattedContent-->` to the files?

gcushen commented:
If you're unfamiliar with Hugo template functions, check out:

- https://github.com/wowchemy/wowchemy-hugo-themes/tree/main/modules/wowchemy/layouts/partials/functions
- https://gohugo.io/templates/partials/#returning-a-value-from-a-partial

Agos95 commented:
I should have applied the requested fixes:
   - moved the code to find the featured image in `partials/functions/get_featured_image.html`
   - use the partial funcion code to get the featured image in the different template files
   - removed the tags `<!--unformattedContent-->` (they were just for my text editor in order to avoid to autoformat those lines on save)

Please, let me know if something more is required!

gcushen reviewed (changes requested):
Please add the line breaks back to the end of files, otherwise it shows as a modification in GitHub and is inconsistent with the rest of the code base.

Agos95 commented:
Ok, I added a blank line at the end of files.

gcushen commented:
@Agos95 thanks for contributing an improvement 🎉
